hazelcast/src/main/java/com/hazelcast/multimap/impl/operations/PutAllBackupOperation.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.multimap.impl.operations;
+
+import com.hazelcast.internal.serialization.Data;
+import com.hazelcast.map.impl.MapEntries;
+import com.hazelcast.multimap.impl.MultiMapContainer;
+import com.hazelcast.multimap.impl.MultiMapDataSerializerHook;
+import com.hazelcast.multimap.impl.MultiMapRecord;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.spi.impl.operationservice.BackupOperation;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Iterator;
+
+public class PutAllBackupOperation extends AbstractMultiMapOperation implements BackupOperation {
+    private transient int currentIndex;
+    private MapEntries mapEntries;
+
+    public PutAllBackupOperation() {
+    }
+
+    public PutAllBackupOperation(String name, MapEntries mapEntries) {
+        super(name);
+        this.mapEntries = mapEntries;
+    }
+
+    @Override
+    public void run() throws Exception {
+        int size = mapEntries != null ? mapEntries.size() : 0;
+        while (currentIndex < size) {
+            Data dataKey = mapEntries.getKey(currentIndex);
+            Data value = mapEntries.getValue(currentIndex);
+            put(dataKey, value);
+            currentIndex++;
+        }
+    }
+
+    protected void put(Data dataKey, Data dataValue) {
```

Comment:
this method needs to be updated similarly to `PutAllOperation` to take into account serialization and also handle properly binary data.